src/utils/runner.ts

```diff
@@ -3,53 +3,120 @@ import { TestRunRequest, TestResult } from "../models/types";
     logger.info("Repository cloned", { repoDir, commitSha });
 
+    // Detect language and get config
     const language = detectLanguage(repoDir);
     const languageConfig = getLanguageConfig(language);
     imageName = `test-image-${commitSha}`;
 
-    // Log content of .hxckr/run.sh, not necessary just for debugging
-    // const runShPath = path.join(repoDir, ".hxckr", "run.sh");
-    // const runShContent = await fs.readFile(runShPath, "utf-8");
+    // Load and write test file
+    const testContent = await loadTestFile(
+      challengeId,
+      language,
+      progress.progress_details.current_step,
+    );
+
+    const appDir = path.join(repoDir, "app");
+    await fs.mkdir(appDir, { recursive: true });
+
+    const testFileName = `stage${progress.progress_details.current_step}${getTestExtension(language)}`;
+    await fs.writeFile(path.join(appDir, testFileName), testContent);
+    logger.info("Test file written", { testFileName });
+
+    // Create run.sh with modified commands
+    const runScript = `#!/bin/bash
+
+    set -e  # Exit on any error
+
+    # Run the tests only (they will execute the code as part of the tests)
+    bun test ./app/stage${progress.progress_details.current_step}${getTestExtension(language)}
+    `;
+
+    const runScriptPath = path.join(repoDir, ".hxckr", "run.sh");
+    await fs.writeFile(runScriptPath, runScript);
+    await fs.chmod(runScriptPath, 0o755); // Make executable
 
     // Build Docker image
     await buildDockerImage(repoDir, imageName, languageConfig.dockerfilePath);
 
     // Run the tests
-    logger.info("Starting test execution", { commitSha });
+    logger.info("Starting test execution", {
+      commitSha,
+      stage: progress.progress_details.current_step,
+    });
+
     const testOutput = await runInDocker(imageName, languageConfig.runCommand);
     logger.info("Test execution completed", { commitSha });
     logger.info("Test output:", { testOutput });
 
+    // Clean and parse the test output
+    const cleanedOutput = cleanTestOutput(testOutput);
+    const success = isTestSuccessful(testOutput);
+
     const testResult: TestResult = {
       event_type: EVENT_TYPE,
-      success:
-        !testOutput.toLowerCase().includes("error") &&
-        !testOutput.toLowerCase().includes("failed"),
-      output: testOutput,
+      repoUrl,
+      commitSha,
+      success,
+      output: cleanedOutput,
     };
 
     await reportResults(commitSha, testResult);
   } catch (error: any) {
     logger.error("Error during test process", { error, commitSha });
+    const errorMessage =
+      error.stderr || error.message || "Unknown error occurred";
     await reportResults(commitSha, {
       event_type: EVENT_TYPE,
+      repoUrl,
+      commitSha,
       success: false,
-      output: "",
-      errorMessage: `Unhandled error: ${error.message}`,
+      output: error.stdout || "",
+      errorMessage: `Test execution failed: ${errorMessage}`,
```

Comment:
The `errorMessage` field is not part of the test result object in line 36. Also, the object response type expected in the backend doesn't include `errorMessage`. Given that we have the error in the output, don't you think having the `errorMessage` field makes it a duplicate?

Reply:
In that case, we should remove the errorMessage field since it's meant to be internal snd not communicated to users

Reply:
Do you see the difference in error report? This is just an example. 

this is when we have the error field (this I prefer)
![image](https://github.com/user-attachments/assets/822318bd-5162-48d7-bc41-fb546dde049f)

and this is when we do not have the error field
![image](https://github.com/user-attachments/assets/af5914bf-3129-4545-bfc8-98fbd483f698)